Bug/6529 class static block

CoyAce opened the pull request:
## :bookmark_tabs: Summary

Brief description about the content of your PR.
disable class static block syntax
Resolves #6529 

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://mermaid.js.org/community/contributing.html)
- [ ] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://mermaid.js.org/community/contributing.html#update-documentation) is used for all new features.
- [ ] :butterfly: If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

sidharthv96 commented on package.json:
```diff
@@ -15,8 +15,10 @@
     "build": "pnpm build:esbuild && pnpm build:types",
-    "build:esbuild": "pnpm run -r clean && tsx .esbuild/build.ts",
+    "build:esbuild": "pnpm clean && tsx .esbuild/build.ts",
+    "build:vite": "pnpm clean && tsx .vite/build.ts",
```
```suggestion
```

Vite build is smaller because it's not building everything esbuild does, and does not have sourcemaps either. 
When compared with the same output (esm), esbuild is smaller than vite (~ half) (and would be much faster, but I just deleted the unrelated files. It's already 10x faster building 4x the things than vite)

<img width="802" alt="image" src="https://github.com/user-attachments/assets/bdb1ff7f-b961-4243-a755-51ba8719e84f" />

```
nr build:vite  69.49s user 7.45s system 172% cpu 44.722 total
nr build:esbuild  6.30s user 1.10s system 85% cpu 8.701 total
```

As we're publishing with esbuild, I don't see a need to keep a vite build in packages, confusing the contributors.

sidharthv96 commented on packages/parser/package.json:
```diff
@@ -8,11 +8,12 @@
   ],
   "homepage": "https://github.com/mermaid-js/mermaid/tree/develop/packages/mermaid/parser/#readme",
+  "module": "./dist/mermaid-parser.esm.mjs",
   "types": "dist/src/index.d.ts",
   "type": "module",
   "exports": {
     ".": {
-      "import": "./dist/mermaid-parser.core.mjs",
+      "import": "./dist/mermaid-parser.esm.mjs",
```
Can you please share an example of why this change is required?

CoyAce replied:
as vite build deprecated, no need to change

sidharthv96 commented on .build/util.test.ts:
```diff
@@ -0,0 +1,15 @@
+import { expect, test } from 'vitest';
+
+import { parseOption } from './util.js';
```
nit
```suggestion
import { expect, test } from 'vitest';
import { parseOption } from './util.js';
```

sidharthv96 commented on .esbuild/util.ts:
```diff
@@ -32,6 +33,8 @@ const buildOptions = (override: BuildOptions): BuildOptions => {
     tsconfig: 'tsconfig.json',
     resolveExtensions: ['.ts', '.js', '.json', '.jison', '.yaml'],
     external: ['require', 'fs', 'path'],
+    supported: { 'class-static-blocks': false },
```
Should this be added only when building with compat mode?

CoyAce replied:
Yes，not necessarily hard coded

sidharthv96 commented on .vite/build.ts:
```diff
@@ -80,7 +80,7 @@ export const getBuildConfig = ({ minify, core, watch, entryName }: BuildOptions)
     },
     resolve: {
-      extensions: [],
+      extensions: ['.js'],
```
Curious why we need this.

sidharthv96 commented on pnpm-lock.yaml:
Can you revert the changes to pnpm-lock, and the vite upgrade? Renovate handles the upgrade for us.

CoyAce replied:
ok

CoyAce replied:
reverted

sidharthv96 reviewed (changes requested):
Thank you for the contribution! 
Have added some comments to discuss about.

sidharthv96 commented:
Chrome <=94 is ~1%, I'm not sure if disabling a feature is worth it for these users, as we use other features like `structuredClone`, which is only supported from 98+

<img width="1507" alt="image" src="https://github.com/user-attachments/assets/3742fe49-5c76-4fba-9a1e-584ff933b98e" />
<img width="1321" alt="image" src="https://github.com/user-attachments/assets/bf44513d-d2c7-46fb-80d2-345a00f0fda0" />

CC @knsv @aloisklink

CoyAce commented:
> Chrome <=94 is ~1% users, and released 4 years back, I'm not sure if disabling a feature is worth it for these users, as we use other features like `structuredClone`, which is only supported from 98+.
> 
> I'd be much happier to approve the PR if we move both the changes in esbuild under the flag.
> 
> <img alt="image" width="1507" src="https://private-user-images.githubusercontent.com/10703445/441872927-3742fe49-5c76-4fba-9a1e-584ff933b98e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY3OTYxMDgsIm5iZiI6MTc0Njc5NTgwOCwicGF0aCI6Ii8xMDcwMzQ0NS80NDE4NzI5MjctMzc0MmZlNDktNWM3Ni00ZmJhLTlhMWUtNTg0ZmY5MzNiOThlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA5VDEzMDMyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMWQwZGJlMzRlN2NlOTVlYmM1MzBmYzRiOTM5NzNlYjdmMDcwYjY2YWQ5NjVkNDk5MzMxNDU1NzQ5YzNlMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4VoRpSzhr3CxA0p7zXYfd3wFZRcbISV8TABiUI8fD5w"> <img alt="image" width="1321" src="https://private-user-images.githubusercontent.com/10703445/441872949-bf44513d-d2c7-46fb-80d2-345a00f0fda0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY3OTYxMDgsIm5iZiI6MTc0Njc5NTgwOCwicGF0aCI6Ii8xMDcwMzQ0NS80NDE4NzI5NDktYmY0NDUxM2QtZDJjNy00NmZiLTgwZDItMzQ1YTAwZjBmZGEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA5VDEzMDMyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2ODZlNTdjNTlkOGNjZTFlMjIwYjNhNTlhZmE1Zjc0MjgzM2M4NmFiMzk3OGE3OTNhN2Q4NmM2OWUzMTZkYTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TUkealNVYzU5VRrNZcawaEHveOFAiXQzfW15O6tOFcY">
> CC @knsv @aloisklink

PR updated, 
1. removed .vite/build.ts because this file is used for bundle, no longer needed
2. reverted vite related change
3. added --supported and --source-map flags
Thanks for your help.
